test/ndb_api_helpers_test.go

```diff
@@ -604,3 +605,50 @@ func TestGenerateProvisioningRequestReturnsErrorIfSSHKeyIsEmpty(t *testing.T) {
 		}
 	}
 }
+
+func TestGetActionArgumentsByDatabaseType(t *testing.T) {
+	// Test with MySQL database type
+	mysqlArgs, err := v1alpha1.GetActionArgumentsByDatabaseType(v1alpha1.DATABASE_TYPE_MYSQL)
+	if mysqlArgs != nil {
+		t.Error("Expected non-nil, but got nil")
+	}
+
+	// Test with Postgres database type
+	postgresArgs, err := v1alpha1.GetActionArgumentsByDatabaseType(v1alpha1.DATABASE_TYPE_POSTGRES)
+	if err != nil {
+		t.Error("Expected non-nil, but got nil")
+	}
+	expectedPostgresArgs := []v1alpha1.ActionArgument{
+		{Name: "proxy_read_port", Value: "5001"},
+		{Name: "listener_port", Value: "5432"},
+		{Name: "proxy_write_port", Value: "5000"},
+		{Name: "enable_synchronous_mode", Value: "false"},
+		{Name: "auto_tune_staging_drive", Value: "true"},
+	}
+	if !reflect.DeepEqual(postgresArgs.GetActionArguments(v1alpha1.DatabaseSpec{Instance: v1alpha1.Instance{DatabaseInstanceName: "test"}}), expectedPostgresArgs) {
+		t.Errorf("Expected %v, but got %v", expectedPostgresArgs, postgresArgs.GetActionArguments(v1alpha1.DatabaseSpec{Instance: v1alpha1.Instance{DatabaseInstanceName: "test"}}))
+	}
+
+	// Test with MongoDB database type
+	mongodbArgs, err := v1alpha1.GetActionArgumentsByDatabaseType(v1alpha1.DATABASE_TYPE_MONGODB)
+	if err != nil {
+		t.Error("Expected non-nil, but got nil")
```

Comment:
"Expected non-nil, but got nil" implies "err == nil" evaluated as "True", but the condition is "err != nil"